File: MVVM/Converters/ThicknessAverageConverter.vb
Imports System.Globalization
Imports System.Windows.Data

Namespace MVVM.Converters

    Public Class ThicknessAverageConverter
        Implements IValueConverter

        Public Function Convert(ByVal Value As Object, ByVal TargetType As Type, ByVal Parameter As Object, ByVal Culture As CultureInfo) As Object Implements IValueConverter.Convert
            Dim Th = CType(Value, Thickness)
            Return (Th.Left + Th.Top + Th.Right + Th.Bottom) / 4
        End Function

        Public Function ConvertBack(ByVal Value As Object, ByVal TargetType As Type, ByVal Parameter As Object, ByVal Culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Dim V = CType(Value, Double)
            Return New Thickness(V)
        End Function

    End Class

End Namespace

File: WPF/MVVM/Converters/ThicknessAverageConverter.cs
using System;
using System.Globalization;
using System.Windows;
using System.Windows.Data;

namespace Ks.Common.MVVM.Converters
{
    public class ThicknessAverageConverter : IValueConverter
    {
        public object Convert(object Value, Type TargetType, object Parameter, CultureInfo Culture)
        {
            var Th = (Thickness) Value;
            return (Th.Left + Th.Top + Th.Right + Th.Bottom) / 4;
        }

        public object ConvertBack(object Value, Type TargetType, object Parameter, CultureInfo Culture)
        {
            var V = System.Convert.ToDouble(Value);
            return new Thickness(V);
        }
    }
}
